tests/posix/common/src/mqueue.c

```diff
@@ -97,4 +96,205 @@ ZTEST(mqueue, test_mqueue)
 	zassert_false(mq_unlink(queue), "Not able to unlink Queue");
 }
 
+bool notification_executed;
+
+void notify_function_basic(union sigval val)
+{
+	mqd_t mqd;
+	bool *executed = (bool *)val.sival_ptr;
+
+	mqd = mq_open(queue, O_RDONLY);
+
+	mq_receive(mqd, rec_data, MESSAGE_SIZE, 0);
+	zassert_false(strcmp(rec_data, send_data),
+		      "Error in data reception. exp: %s act: %s", send_data, rec_data);
+
+	zassert_false(mq_close(mqd),
+		      "Unable to close message queue descriptor.");
+
+	*executed = true;
+}
+
+ZTEST(mqueue, test_mqueue_notify_basic)
+{
+	mqd_t mqd;
+	struct mq_attr attrs;
+	struct sigevent not;
+	int32_t mode = 0777;
+	int flags = O_RDWR | O_CREAT;
+
+	attrs.mq_msgsize = MESSAGE_SIZE;
+	attrs.mq_maxmsg = MESG_COUNT_PERMQ;
+
+	not.sigev_notify = SIGEV_NONE;
+	not.sigev_value.sival_ptr = (void *)&notification_executed;
+	not.sigev_notify_function = notify_function_basic;
+	not.sigev_notify_attributes = NULL;
```

Comment:
Same - just a suggestion, not blocking, so you can ignore if there is no respin required.
```suggestion
	mqd_t mqd;
	struct mq_attr attrs = {
		.mq_msgsize = MESSAGE_SIZE,
		.mq_maxmsg = MESG_COUNT_PERMQ,
	};
	struct sigevent not = {
		.sigev_notify = SIGEV_NONE,
		.sigev_value.sival_ptr = (void *)&notification_executed,
		.sigev_notify_function = notify_function_basic,
	};
	int32_t mode = 0777;
	int flags = O_RDWR | O_CREAT;
```